dev/NavigationView/NavigationView.cpp

```diff
@@ -714,6 +714,7 @@ void NavigationView::RaiseItemInvokedForNavigationViewItem(const winrt::Navigati
         auto inspectingDataSource = static_cast<InspectingDataSource*>(winrt::get_self<ItemsSourceView>(itemsSourceView));
         auto itemIndex = parentIR.GetElementIndex(nvi);
         nextItem = inspectingDataSource->GetAt(itemIndex);
+        SelectedItem(nextItem);
```

Comment:
I was expecting the order to be incorrect, but it looks like we are selecting on invoke which we were not doing before ?

Reply:
Yes, correct. The order wasn't wrong, the problem is that the method calling the `RaiseItemInvokedForNavigationViewItem` does not change selection at all, and `SelectedItem` returns the old selection, not the item we should be focusing now.

Reply:
Update: Turns out the order inside OnNavigationViewItemInvoked was probably the root cause of the issue, sorry about the confusion.